vali_objects/utils/elimination_manager.py

```diff
@@ -69,6 +69,29 @@ def _delete_eliminated_expired_miners(self):
             if hotkey in self.metagraph.hotkeys:
                 bt.logging.trace(f"miner [{hotkey}] has not been deregistered by BT yet. Not deleting miner dir.")
                 continue
+
+            # Now to manage the deletion of the miner's challengeperiod information if it exists
+            try:
+                # Collect existing challengeperiod information from disk
+                self._refresh_challengeperiod_in_memory()
```

Comment:
can just do this once before the loop over all hotkeys. reading from disk is expensive.

Reply:
updated